torch/csrc/Module.cpp

```diff
@@ -1462,6 +1469,9 @@ PyObject* initModule() {
 #ifdef USE_CUDA
   THPUtils_addPyMethodDefs(methods, THCPModule_methods());
 #endif
+#ifdef USE_XPU
+  THPUtils_addPyMethodDefs(methods, THXPModule_methods());
+#endif
```

Comment:
Just curious, why we want new TPHXP modules?

Reply:
We declare the C++ impl in `THXPModule_methods` and bind them to python APIs here.

Reply:
TH references old method of implementing kernels, which heavily relied on templates. If I'm not mistaken, we finally got rid of it, but it still exists in one or two places In the codebase

Reply:
Got your point. Use `torch::xpu::python_functions` to replace the old code style.